src/admin/render_readmes.rs

```diff
@@ -111,25 +110,10 @@ pub fn run(opts: Opts) -> anyhow::Result<()> {
             let handle = thread::spawn::<_, anyhow::Result<()>>(move || {
                 println!("[{}-{}] Rendering README...", krate_name, version.num);
                 let readme = get_readme(base_config.uploader(), &client, &version, &krate_name)?;
-                let content_length = readme.len() as u64;
-                let content = std::io::Cursor::new(readme);
-                let readme_path = format!("readmes/{0}/{0}-{1}.html", krate_name, version.num);
-                let mut extra_headers = header::HeaderMap::new();
-                extra_headers.insert(
-                    header::CACHE_CONTROL,
-                    header::HeaderValue::from_static(CACHE_CONTROL_README),
-                );
 
                 base_config
                     .uploader()
-                    .upload(
-                        &client,
-                        &readme_path,
-                        content,
-                        content_length,
-                        "text/html",
-                        extra_headers,
-                    )
+                    .upload_readme(&client, &krate_name, &version.num, readme)
```

Comment:
`upload_readme()` appears to use a different `Cache-Control` header, if I understand this correctly. admittedly, when we upload first via `upload_readme()` with an immutable cache policy then rerendering and uploading with a different header appears a bit pointless, but it still looks like a potentially significant change. do you think you can extract that refactoring into a dedicated pull request so that we can merge and deploy it separately? :)